kotlin/internal/jvm/compile.bzl

```diff
@@ -137,6 +137,21 @@ def _partition_srcs(srcs):
         src_jars = depset(src_jars),
     )
 
+def _handle_legacy_friends(ctx):
+    # TODO extract and move this into common. Need to make it generic first.
+    friend = getattr(ctx.attr, "friend", None)
+
+    # Reduce to a single friend if more than one is set (old API)
+    friends = getattr(ctx.attr, "friends", [])
+    if len(friends) == 1:
+        if bool(friend):
+            fail("Can only use friend or friends, but not both. Please only set `friend=`")
```

Comment:
I think this should handle situation when you set `friend` and multiple `friends` (not just one).

Reply:
If you look down one block, setting multiple friends is already disallowed, regardless of whether you used `friend=`.  That was already true in the old code, I'm just re-doing it here. but if you use `friends=` correctly (despite it being deprecated), this second check just makes sure you're not doing both.